Precompilation errors with Trixi, SciMLBase

Current compilation gives errors like
```bash
Failed to precompile Trixi [a7f1ee26-1774-49b1-8366-f1abc58fbfcb] to "C:\\Users\\huiyu\\.julia\\compiled\\v1.10\\Trixi\\jl_B0A6.tmp".
ERROR: LoadError: AssertionError: Base.precompile(Tuple{DiscreteCallback{typeof(Trixi.summary_callback), typeof(Trixi.summary_callback), typeof(Trixi.initialize_summary_callback), typeof(SciMLBase.FINALIZE_DEFAULT), typeof(nothing)}})
```
and the fix is related to https://github.com/trixi-framework/Trixi.jl/pull/2549.

Current PRs are stalled by this issue: #184, #183, and an upcoming PR to bump Trixi to the latest version.

--- a/.github/workflows/Documenter.yml
+++ b/.github/workflows/Documenter.yml
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
 
       - name: Set up Julia 1.10
         uses: julia-actions/setup-julia@v2

--- a/.github/workflows/FormatCheck.yml
+++ b/.github/workflows/FormatCheck.yml
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
 
       - name: Run Julia Format Check
         uses: julia-actions/julia-format@v4

--- a/.github/workflows/SpellCheck.yml
+++ b/.github/workflows/SpellCheck.yml
@@ -8,9 +8,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
 
       - name: Run Typo and Spelling Check
-        uses: crate-ci/typos@v1.34.0
+        uses: crate-ci/typos@v1.35.7
         with:
           config: .typos.toml # Specify the custom configuration file

--- a/Project.toml
+++ b/Project.toml
@@ -16,7 +16,7 @@ CUDA = "5"
 SciMLBase = "2"
 StaticArrays = "1"
 Test = "1.10"
-Trixi = "0.11 - 0.12"
+Trixi = "0.11 - 0.13.5"
 julia = "1.10"
 
 [extras]

